### ISSUE TITLE
Two versions of the same object are rendered

### ISSUE THREAD

**Issue by contrapunctus-1** [osmandapp/Osmand#5789]
OsmAnd v3.0.4 from F-Droid on a Nokia 6.1 running Android v8.1

![](https://i.imgur.com/jElCnAD.png)

![](https://i.imgur.com/wicwKxW.png)

![](https://i.imgur.com/VZXBADn.png)

I've tried clearing the app's cache, and deleting the map data and downloading it again.

The place pictured above is NH205, Himachal Pradesh, India
Location: geo:31.2304,76.66127?z=16
https://osmand.net/go?lat=31.2304&lon=76.66127&z=16

**Comment by gunnar22** [comment]
I had the same issue some time ago and found a solution. I had at the same time both a map for the whole country (Norway) and for a part of the country. One of them I had updated. OsmAnd displayed both maps (the big one and the small one) on top of each other, so I saw at the same time both old and a new versions of the map details. When I understood the reason, I closed OsmAnd and deleted the old maps from the file explorer in Android. This solved the problem. 
(I did not measure time usage, but probably it was slower doing rendering due to showing two maps instead of one all the time.)
My suggestion is that you investigate if you have the same situation with the map files.

/gunnar22

**Comment by contrapunctus-1** [comment]
Thanks for chiming in, gunnar22. I deleted all .obf files in Android/data/net.osmand.plus/files/ , then downloaded the following regions, as before -
1. Delhi
2. Himachal Pradesh
3. Punjab
4. Haryana
5. Chandigarh
6. World overview map

(So it doesn't look like there's any overlap in this case, except for Chandigarh, which I haven't looked at.)

Now the region in question rendered only the old data.

I then enabled live updates and updated all regions, and the problem was reproduced - both old and new versions were rendered.

**Comment by naoliv** [comment]
Do you still see this problem if you disable only the world basemap?

**Comment by contrapunctus-1** [comment]
@naoliv yes.

**Comment by PaulStets** [comment]
@contrapunctus-1 Could you please try to reproduce this now?
I have tried to reproduce the issue with and without the live updates and was not able to.
![image](https://user-images.githubusercontent.com/20681525/45758299-f7199800-bc2d-11e8-8f69-7adaa1af1011.png)

**Comment by contrapunctus-1** [comment]
@PaulStets Great, I can't either. \o/

**Comment by PaulStets** [comment]
@contrapunctus-1 Sorry, but I still cannot reproduce this both with and without the live updates. (see attached screenshots). Please backup OsmAnd's data folder and reinstall the app.
![image](https://user-images.githubusercontent.com/20681525/45997097-30775b00-c0a7-11e8-9a2c-e691114eefd9.png)
![image](https://user-images.githubusercontent.com/20681525/45997102-35d4a580-c0a7-11e8-8818-bf220f8f438d.png)

**Comment by scaidermern** [comment]
Maybe emptying (or renaming) the `live` directory is enough to solve this problem? There has been a problem in the past with old differential updates not being deleted.

**Comment by PaulStets** [comment]
Perhaps this could help. However, if the issue persists, I would still recommend reinstalling the app, as the issue may have something to do with the rendering engine.

**Comment by contrapunctus-1** [comment]
When I looked at it again just now, the duplicate object was no longer there. Didn't get a chance to try the `live` directory move as suggested above.

@vscherb if possible, please leave this issue open until it is resolved at the source.

**Comment by vshcherb** [comment]
I think we have similar issue open anyway we've updates for closed issues too the most important criteria for us is to be able to reproduce, unfortunately it is not the case somehow. Once we understand what's going on we will fix it

**Comment by Bubu** [comment]
I just experienced this with some backcountry trails in Alberta, Canada.

I updated the OSM data yesterday and this morning received the update via OSMAnd Live. Checking the area I say two versions of the trail rendered. Closing and restarting osmand didn't change anything. What solved the problem is disabling OSMAnd live for alberta and then enabling it again.

This is with OSMAnd 3.2.0 from F-Droid. I only have Alberta downloaded for ~3 weeks now and have daily updates enabled.
